compiler/testData/codegen/box/operatorConventions/assignOperatorOverload.kt

```diff
@@ -45,10 +86,27 @@ fun box(): String {
     foo.x[1] = 2L
     if (foo.x.value != "OK.Container.set2") return "Fail: ${foo.x.value}"
 
-    // Test assign() on null is not called
+    // Test operator String.assign is selected over SelectAssignTest.String.assign
+    result = "Fail"
+    SelectAssignTest().test()
```

Comment:
It would be great to have a test where `operator assign` is a class method, not an extension, and we resolve correctly to this class method.

Reply:
Added additional tests for that case.